axelor-sale/src/main/java/com/axelor/apps/sale/service/saleorder/SaleOrderService.java

```diff
@@ -119,4 +120,11 @@ public List<SaleOrderLine> handleComplementaryProducts(SaleOrder saleOrder)
   SaleOrder separateInNewQuotation(
       SaleOrder saleOrder, ArrayList<LinkedHashMap<String, Object>> saleOrderLines)
       throws AxelorException;
+
+  public void createNewVersion(SaleOrder saleOrder);
+
+  public LocalDateTime getVersionDateTime(SaleOrder saleOrder, Integer versionNumber);
+
+  public boolean recoverVersion(
+      SaleOrder saleOrder, Integer versionNumber, boolean saveActualVersion);
```

Comment:
Please instead move these new methods to a new Service `SaleOrderVersionService`